wlroots/wlr_types/compositor.py

```diff
@@ -21,3 +28,135 @@ def __init__(self, display: Display, renderer: Renderer) -> None:
             The wlroots renderer to attach the compositor to.
         """
         self._ptr = lib.wlr_compositor_create(display._ptr, renderer._ptr)
+
+
+class SubCompositor(Ptr):
+    def __init__(self, display: Display) -> None:
+        self._ptr = lib.wlr_subcompositor_create(display._ptr)
+
+
+class Surface(PtrHasData):
+    def __init__(self, ptr):
+        """Create a wlroots Surface
+
+        :param ptr:
+            The cdata for the given surface
+        """
+        self._ptr = ptr
+
+        self.commit_event = Signal(ptr=ffi.addressof(self._ptr.events.commit))
+        self.new_subsurface_event = Signal(
+            ptr=ffi.addressof(self._ptr.events.new_subsurface),
+            data_wrapper=SubSurface,
+        )
+        self.destroy_event = Signal(ptr=ffi.addressof(self._ptr.events.destroy))
+
+    def __eq__(self, other: object) -> bool:
+        """Check if the other surface is equal by the cdata pointer"""
+        if not isinstance(other, Surface):
+            return NotImplemented
+
+        return self._ptr == other._ptr
```

Comment:
Is there a reason to need a different version than what exists in `Ptr.__eq__`? Should we change that method to do something more like this?

Reply:
Good question, I can't remember why I wrote this method. Hmm. Nothing breaks in https://github.com/qtile/qtile/pull/3985 when I remove it so I was likely debugging and left it in.